patches/tModLoader/Terraria/ModLoader/ModdedPylonsMapLayer.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.Xna.Framework;
+using System.Collections.Generic;
+using Terraria.GameContent;
+using Terraria.Map;
+
+namespace Terraria.ModLoader
+{
+	internal class ModdedPylonsMapLayer : IMapLayer
```

Comment:
If you make this `ModMapLayer` then you don't have to edit `Main.MapLayers`

Additionally... can we not combine this into vanilla `TeleportPylonsMapLayer`? I think `ModPylon.DrawMapIcon` should have an easier to use implementation where they just return the texture/color, and it's otherwise fully vanilla drawing.